sympy/ntheory/modular.py

```diff
@@ -253,3 +253,39 @@ def combine(c1, c2):
         if symmetric:
             return symmetric_residue(n, m), m
         return n, m
+
+
```

Comment:
The doctest is not being tested, currently. Some friendly suggestions (which don't include changing the arg list) are:
```diff
diff --git a/sympy/ntheory/modular.py b/sympy/ntheory/modular.py
index e5b14a9..92ce21a 100644
--- a/sympy/ntheory/modular.py
+++ b/sympy/ntheory/modular.py
@@ -257,35 +257,36 @@ def combine(c1, c2):
 
 def crt_cartesian(rem, mod):
     """
-    Crt over a cartesian product of vectors.given a vector of moduli,
-    and for each modulus given a vector of remainders, instead of usual one remainder in crt.
-    It outputs solutions that leave a remainder matching one of the remainders
-    of the vector of remainders corresponding to a modulus.
-    Assuming that moduli are pairwise relatively prime.
+    Return a list of values whose elements give the remainders
+    in the Cartesian product of the supplied remainders when divided
+    by each of the supplied moduli. None is returned if the moduli
+    are not pairwise-prime.
+
     Examples
     ========
+
     >>> from sympy.ntheory.modular import crt_cartesian
-    >>> crt_cartesian([[3,5], [3,7]], [7, 11])
+    >>> from sympy.utilities.iterables import cartes
+    >>> rem = [[3, 5], [3, 7]]
+    >>> crt_cartesian(rem, [7, 11])
     [3, 73, 47, 40]
-    >>> [(p%7, p%11) for p in [3, 40, 47, 73]]
-    [(3, 3), (5, 7), (5, 3), (3, 7)]
+    >>> [(p%7, p%11) for p in _]
+    [(3, 3), (3, 7), (5, 3), (5, 7)]
+    >>> assert _ == list(cartes(*rem))
+
     """
-    if len(mod) > len(rem):
-        raise ValueError("Too few remainders")
     if len(mod) == 0:
-        raise ValueError("Moduli Vector can't be empty")
+        raise ValueError("There must be at least 1 modulus.")
+    if len(mod) != len(rem) or not all(rem):
+        raise ValueError(
+            "There should be a list of remainders for each modulus.")
     m = mod[0]
     R = rem[0]
-    for i in range(1,len(mod)):
-        rem2 = []
-        s = mod_inverse(m, mod[i])
-        _m = m
+    for i in range(1, len(mod)):
+        try:
+            mi = m*mod_inverse(m, mod[i])
+        except ValueError:
+            return None  # mod_inverse fails if values are not coprime
         m *= mod[i]
-        for elem in R:
-            for k in rem[i]:
-                r = elem
-                r += _m*s*(k - r)
-                r %= m
-                rem2.append(r)
-        R = rem2
+        R = [(r + mi*(ri - r)) % m for r in R for ri in rem[i]]
     return R
```

Reply:
```R = [(r + mi*(ri - r)) % m for r in R for ri in rem[i]]``` this gave incorrect results.
I tried to get the correct expression and found something like below:
 ```
        prev_m = m
        m *= mod[i]
        R = [(r + prev_m*mi*(ri - r)) % m for r in R for ri in rem[i]]
```

Reply:
Yes, I left off the `m` from the `mi` caluclation.